### DISCUSSION TITLE
How do I access a string from interactive.string function?

### DISCUSSION THREAD

**Discussion by Russsgithub** [savonet/liquidsoap#2061] [User Support]
I'm trying to use interactive.string but get a ``` fun () -> (_) not string ``` error in some situations.

Using this code -

```
string = interactive.string("string", "this is a string")

print("#{string}")
```

output of print is ```<fun>```

```
string = interactive.string("string", "this is a string")

print(string)
```

output of print is ```<fun>.{set = <fun>, set = <fun>, remove = <fun>}```

How do i get a string from interactive.string ?

Thanks

**Comment by Russsgithub** [comment]
A version would be useful I guess, sorry.

I'm using the latest code form github in the above example.

Thanks

**Comment by smimram** [comment]
I'd say
```
string = interactive.string("string", "this is a string")

print(string())
```
An interactive something is a function which takes no argument and returns the something.

**Reply by Russsgithub** [reply]
Thanks @smimram , that'd not even crossed my mind.
